### PR TITLE
Fix Crossgen2 PDB generation (again)

### PR DESCRIPTION
**Pull request by trylek**

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Diagnostics/PdbWriter.cs [hunk 1]
```diff
@@ -90,6 +90,7 @@ public partial class PdbWriter
         Dictionary<SymDocument,int> _documentToChecksumOffsetMapping;
 
         UIntPtr _pdbMod;
+        IntPtr _pdbWriterInst;
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Diagnostics/PdbWriter.cs:93]: I think you can release the reference right after the GetOrCreateObjectForComInstance call. No need to store it in an extra instance field.

Comment (trylek) [reply src/coreclr/tools/aot/ILCompiler.Diagnostics/PdbWriter.cs]: Good point, thanks Jan for pointing that out, fixed in 2nd commit.

**Comment by jkotas** [comment]
This is regression introduced by removal of the Dispose call here: https://github.com/dotnet/runtime/pull/84643/files#diff-9e985fc04e2df61f33c3d19bb5d1a0de0188a8343339bfc2f8c977c56d1f60f2L148

@jtschuster @jkoritzinsky Is this change the best fix?

**Review by jkoritzinsky — APPROVED** [review]

**Review by davidwrighton — APPROVED** [review]
Looks good to me. Please run this through the process to backport to 8.0, since this is a regression.

**Pull request merged**